### PR TITLE
Add description to Attachments and Files

### LINKED ISSUE #483: Add description to attachments
The description property is missing in attachments (https://docs.pycord.dev/en/master/api.html?highlight=attachment#discord.Attachment).

It is supported and used for screenreader / alt text.
See https://discord.com/developers/docs/resources/channel#attachment-object-attachment-structure

The field is a optional string.

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

Adds the `description` attribute to `Attachment` and `File`

@zeffo helped with implementing file sending.

Resolves #483

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by plun1331** [comment]
Tried to add to interaction message edits but after a half hour couldn't figure out what i was doing

**Comment by Lulalaby** [comment]
> Tried to add to interaction message edits but after a half hour couldn't figure out what i was doing

Interactions seems to be broken for descriptions

**Review by BobDotCom — CHANGES_REQUESTED** [review]
Why is single file support removed from this?

File: discord/message.py [hunk 1]
```diff
@@ -1319,6 +1324,9 @@ async def edit(
             raise InvalidArgument('cannot pass both file and files parameter to edit()')
 
         if file is not MISSING:
+            if 'attachments' not in payload:
+                # don't want it to remove any attachments when we just add a new file
```
[the remark this answers is not in the document]
Comment (Lulalaby) [reply discord/message.py:1328]: This btw.

[the remark this answers is not in the document]
Comment (plun1331) [reply discord/message.py:1331]: You can still remove files the same way, this just makes is so if you don't explicitly pass in attachments then it won't remove them.

File: discord/file.py [hunk 2]
```diff
@@ -61,22 +61,26 @@ class File:
         The filename to display when uploading to Discord.
         If this is not given then it defaults to ``fp.name`` or if ``fp`` is
         a string then the ``filename`` will default to the string given.
+    description: Optional[:class`str`]
+        A description of the file, used by Discord to display alt text on images.
```
Comment (RPSMain) [inline discord/file.py:65]:
```suggestion
        The description of a file, used by Discord to display alternative text on images.
```

File: discord/interactions.py [hunk 3]
```diff
@@ -683,7 +683,7 @@ async def edit_message(
                 payload['components'] = []
             else:
                 payload['components'] = view.to_components()
-
+
```
Comment (RPSMain) [inline discord/interactions.py:686]:
```suggestion

```

**Review by RPSMain — CHANGES_REQUESTED** [review]

File: discord/message.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BobDotCom) [reply discord/message.py:1331]: The ideal functionality would be:
```py
m = await send("hi", file=x)  # Sends with file x
await m.edit("hello")  # Leaves file x
await m.edit(files=m.attachments+[y])  # Adds file y
await m.edit(files=[y])  # Removes file x
await m.edit(files=[])  # Removes files
```
I'm not able to read through all your code again at the moment. Does your code function in this way?

[the remark this answers is not in the document]
Comment (plun1331) [reply discord/message.py:1331]: The usage should look like this:
```py
m = await send("hi", file=x)  # Sends with file x
await m.edit("hello")  # Leaves file x
await m.edit(files=[y])  # Adds file y
await m.edit(attachments=[y])  # Removes file x, y here would be a discord.Attachment
await m.edit(attachments=[])  # Removes files
```

Here's the code I actually used for testing:
```py
msg = await message.reply('a', files=[
    discord.File('banner.jpg', description='Profile banner'), # banner 1
    discord.File('banner.jpg', description='Profile banner 2') # banner 2
]) # sends message with 2 attachments
await asyncio.sleep(3)
msg = await message.edit('e') # leaves both attachments
file = discord.File('banner.jpg', description='Profile banner 3') # banner 3
attachment = msg.attachments[0] 
await asyncio.sleep(3)
await msg.edit(file=file, attachments=[attachment]) # adds banner 3, removes banner 2
```

I tried to keep the existing behavior of `edit`, where `attachments` is used to specify what files to keep, and `files` is used to add new files.

**Review by RPSMain — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Pull request merged**